build.prod.js

```diff
@@ -1,24 +1,70 @@
-async function build() {
-    const chalk = (await import("chalk")).default
-
-
-    require('esbuild').build({
-        entryPoints: ['client/client.ts'],
-        bundle: true,
-        outfile: 'dist/client.js',
-        target: ["chrome58"],
-        minify: false,
-        format: "iife",
-    }).then(() => console.log(chalk.green("[client]: Built successfully!"))).catch(() => process.exit(1))
-
-    require('esbuild').build({
-        entryPoints: ['server/server.ts'],
-        bundle: true,
-        outfile: 'dist/server.js',
-        format: "cjs",
-        minify: false,
-        platform: "node",
-    }).then(() => console.log(chalk.green("[server]: Built successfully!"))).catch(() => process.exit(1))
+const fs = require('fs');
+const path = require('path');
+const { build } = require('esbuild');
+
+function getAllFiles(folder) {
```

Comment:
not sure what we benefit from this

Reply:
since entry points refer to where your program entries, and that is 1 file, the rest of the files will get hooked by esbuild resolving it from the import statements

Reply:
While true, ESBuilds module resolution does **not** account for directory structures -- it will just dump them into the root of where the resolved module resides. For example, if I have a file structure like this:
```
server/
   file_a.ts
   some_other_folder/
    file_b.ts
 ```
 
 And imported `file_b.ts` from within `file_a.ts` - the `dist` output would be:
 ```
 dist/
  file_a.js
  file_b.js // ! structure removed
```

This destroys the structure we are using in our project, which can be bad for several reasons, mainly consistency. Even though it's built code and no one will be touching it, it's still good to keep things the same as much as possible.

Reply:
This approach can definitely lead to pitfalls, since eg file_b is not an entry point to our program, and other files depend on it.

Reply:
Reverted in [019b5cc](https://github.com/wowjeeez/esbuild-fivem/pull/2/commits/019b5cc4ff21a27acb6d7fbb152c7991329c11cc)